src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java

```diff
@@ -262,6 +261,31 @@ public InnReachTransactionDTO cancelPatronHold(UUID transactionId, CancelPatronH
     return transactionMapper.toDTO(transaction);
   }
 
+  @Override
+  public void returnPatronHoldItem(UUID transactionId, UUID servicePointId) {
+    var transaction = fetchTransactionOfType(transactionId, PATRON);
+
+    verifyState(transaction, ITEM_RECEIVED, RECEIVE_UNANNOUNCED);
+
+    log.info("Attempting to return item of Patron Hold transaction {}", transactionId);
+
+    var loanId = transaction.getHold().getFolioLoanId();
+    var requestId = transaction.getHold().getFolioRequestId();
+    if (loanId != null) {
+      var loan = loanService.getById(loanId);
+      if (loanService.isOpen(loan)) {
+        log.info("Checking-in item for transaction associated with an open loan");
+        loanService.checkInItem(transaction, servicePointId);
+      }
+    } else {
+      var request = requestService.findRequest(requestId);
+      if (requestService.isCanceledOrExpired(request)) {
+        log.info("Checking-in item for transaction associated with a closed loan and canceled request");
+        loanService.checkInItem(transaction, servicePointId);
```

Comment:
Shouldn't the RETURN_UNCIRCULATED message be sent?

Reply:
This logic will trigger Check-In kafka event and the message will be sent there